### PR TITLE
[Merged by Bors] - feat: port AlgebraicTopology.DoldKan.Compatibility

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
@joelriou The corresponding file in mathlib seems to be actively worked on, and I have trouble building it, hence it is hard to fix the last two remaining proofs: `equivalence₁CounitIso_eq` and `equivalence₁UnitIso_eq`, please have a look.

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean [hunk 1]
```diff
@@ -51,145 +51,149 @@ namespace DoldKan
...
+  simp only [equivalence₂UnitIso_hom_app, equivalence₁UnitIso_eq, equivalence₁UnitIso_hom_app,
+      assoc, NatIso.cancel_natIso_hom_left]
+  rfl
 #align algebraic_topology.dold_kan.compatibility.equivalence₂_unit_iso_eq AlgebraicTopology.DoldKan.Compatibility.equivalence₂UnitIso_eq
 
 variable {eB}
 
-include hG
+-- porting note: commented out include
+-- include hG
```
Comment (semorrison) [inline Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:180]: Just remove.

Comment (semorrison) [reply Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:180]: (and the other two)

**Comment by semorrison** [comment]
bors d+

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+/-- A basic equivalence `A ≅ B'` obtained by composing `eA : A ≅ A'` and `e' : A' ≅ B'`. -/
+@[simps! functor inverse unitIso_hom_app]
+def equivalence₀ : A ≌ B' :=
+  eA.trans e'
+#align algebraic_topology.dold_kan.compatibility.equivalence₀ AlgebraicTopology.DoldKan.Compatibility.equivalence₀
+
+-- porting note: commented out include
+-- include hF
+
```
Comment (semorrison) [inline Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:65]:
```suggestion
```

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  ext X
+  dsimp [equivalence₁]
+  unfold Functor.asEquivalence
+  dsimp [NatIso.hcomp, IsEquivalence.ofEquivalence]
+  simp
+#align algebraic_topology.dold_kan.compatibility.equivalence₁_unit_iso_eq AlgebraicTopology.DoldKan.Compatibility.equivalence₁UnitIso_eq
+
+-- porting note: commented out include
+-- include eB
+
```
Comment (semorrison) [inline Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:124]:
```suggestion
```

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  dsimp [equivalence₂]
+  simp only [equivalence₂UnitIso_hom_app, equivalence₁UnitIso_eq, equivalence₁UnitIso_hom_app,
+      assoc, NatIso.cancel_natIso_hom_left]
+  rfl
+#align algebraic_topology.dold_kan.compatibility.equivalence₂_unit_iso_eq AlgebraicTopology.DoldKan.Compatibility.equivalence₂UnitIso_eq
+
+variable {eB}
+
+-- porting note: commented out include
+-- include hG
```
Comment (semorrison) [inline Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:180]:
```suggestion
```

**Comment by semorrison** [comment]
bors merge

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean]: make sense..
I saw you already did it just now

**Pull request closed without merging**